dissect/target/plugins/os/windows/log/mssql.py

```diff
@@ -89,12 +89,12 @@ def errorlog(self) -> Iterator[MssqlErrorlogRecord]:
 
                     buf += line
 
-    def _find_instances(self) -> list[str, TargetPath]:
-        instances = []
+    def _find_instances(self) -> set[str, TargetPath]:
+        instances: set = set()
 
-        for subkey in self.target.registry.key(self.MSSQL_KEY).subkeys():
-            if subkey.name.startswith("MSSQL") and "." in subkey.name:
-                instances.append(
+        for subkey in self.target.registry.glob_ext(self.MSSQL_KEY_GLOB):
```

Comment:
I think using `keys` instead of `key` also would've resolved this.

Reply:
Thats correct. Would you prefer that? In this case the globing works nice with the MSSQL<pattern>.<instance_name> subkeys.

Reply:
I think since you're doing additional filtering after the glob anyway, might as well check everything without the glob. Probably a bit faster too.

If you could do this with just a glob, then that'd probably be nicer.